pages/api/databases/upload.ts

```diff
@@ -0,0 +1,81 @@
+import type { NextApiRequest, NextApiResponse } from 'next';
+import { getUser, run } from '../../../lib/apiUtils';
+import { IncomingForm, Fields, Files, File } from 'formidable';
+import fs from 'fs';
+import { connection, connectToDatabase } from '../../../lib/database';
+
+// first we need to disable the default body parser
+export const config = {
+  api: {
+    bodyParser: false,
+  },
+};
+
+const testuserPassword = 'EC62DEF08F5E4FD556DAA86AEC5F3FB0390EF8A862A41ECA';
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  const user = await getUser(req, res);
+  if (typeof user === 'undefined') return;
+
+  await connectToDatabase();
+
+  const data = (await new Promise((resolve, reject) => {
+    const form = new IncomingForm();
+
+    form.parse(req, (err, fields, files) => {
+      if (err) return reject(err.toString());
+      resolve({ fields, files });
+    });
+  })) as
+    | {
+        readonly fields: Fields;
+        readonly files: Files;
+      }
+    | string;
+
+  if (typeof data === 'string') return res.status(400).json({ error: data });
+
+  const databaseName = data.fields.databaseName as string | undefined;
+
+  if (!databaseName)
+    return res.status(400).json({ error: 'Database name is required' });
+
+  if (databaseName.match(/^\w+$/) === null)
+    return res.status(400).json({ error: 'Database name is invalid' });
+
+  const filePath = (data.files.file as File | undefined)?.path;
+
+  if (typeof filePath === 'undefined')
+    return res.status(400).json({ error: 'No file is attached' });
+
+  await run(`sed -i -e 's/^CREATE DATABASE.*$//g' ${filePath}`)
+    .then(() => run(`sed -i -e 's/^USE .*$//g' ${filePath}`))
+    .then(() => connection.execute(`DROP DATABASE IF EXISTS \`${databaseName}\``))
```

Comment:
You should see if your connection library supports parameters here. This is susceptible to SQL injection. Not a big deal for internal use, but if the library supports it might as well use it. Also it would avoid the whole quoting issue.

Reply:
The library supports templating, but not everywhere, If I am correct. I don't think you can template a database name.

Still, this case should not create an SQL injection thread as I check database name against regex on line 46:
```js
if (databaseName.match(/^\w+$/) === null)
```
and, the `filePath` is generated by the system, not the user. The format is always something like `/tmp/upload_2h934htkj2h3t4j`

Reply:
Just checked again.
5a84a06
The library does not support templated database name